### PR TITLE
fix: (core) DatePicker - fix form's valid flag, when allowNull

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
fixes: https://github.com/SAP/fundamental-ngx/issues/1598
#### Please provide a brief summary of this pull request.
Right now I check if the input is empty and allowNull flag is added. Also there was some minor bug on datepicker, which is fixed now.
#### Please check whether the PR fulfills the following requirements
**This is only fix PR, without any new things. So no docs needs to be changed.**
- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples

### CODE REVIEW

**Comment by droshev** [comment]
@JKMarkowski can you put `(core)` after `:`

File: libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts [hunk 1]
```diff
@@ -212,20 +212,22 @@ export class CalendarDayViewComponent implements OnInit, OnChanges, OnDestroy {
 
     /** @hidden
      *  Amount of selected days
-     *  0, when none,
-     *  1, when only startDate, or endDate same as startDate,
-     *  2, when both
+     *  0, when none of selected, or start date is invalid,
```
Comment (InnaAtanasova) [inline libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts]: My English is not good, but doesn't this mean exactly the same as point 2? (both are valid)
@stefanoScalzo some help here?

File: libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts [hunk 2]
```diff
@@ -212,20 +212,22 @@ export class CalendarDayViewComponent implements OnInit, OnChanges, OnDestroy {
...
+     *  1, when only startDate, or endDate same as startDate, but startDate is valid,
```
Comment (InnaAtanasova) [inline libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts]: And here I am lost ...

File: libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts (hunk 1 above)
Comment (stefanoScalzo) [reply libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts]: should be
0, when selected and startDate are valid
1, when given valid startDate and endDate is same as startDate

File: libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts (hunk 2 above)
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts]: It has been changed :)

File: libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts (hunk 1 above)
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts]: It has been changed, please take a look on it now.

**Comment by InnaAtanasova** [comment]
Perfect!

**Pull request merged**